pyccel/codegen/pipeline.py

```diff
@@ -238,13 +238,8 @@ def handle_error(stage):
     if semantic_only:
         return
 
-    if parser.module_parser:
-        parsers = [parser.module_parser, parser]
-        program_name = os.path.basename(os.path.splitext(parser.filename)[0])
-        module_names = [module_name, program_name]
-    else:
-        parsers = [parser]
-        module_names = [module_name]
+    parsers = [parser]
```

Comment:
is it necessary to use list, we have only one item.
we may change also the name to `parser`

Reply:
No, it is not necessary, but have fun reviewing the changes that come with getting rid of it ^^. See 14683d69362a2987b6534c0252ae63646553b3e9
I just removed these 2 lines and got rid of the loop, but it's going to look like a huge change. Don't feel like you need to check every line XD

Reply:
great, sometimes we need huge changes to make things better.